packages/identity-app/src/SavedAccounts.tsx

```diff
@@ -32,27 +50,21 @@ export class SavedAccounts extends React.PureComponent<Props> {
   }
 
   renderAllAccountsFromKeyring () {
-    const { keyring } = this.context;
+    const { allAccounts } = this.state;
 
-    return (
-      <React.Fragment>
-        {
-          keyring.getPairs().map(pair => {
-            return (
-              <React.Fragment key={pair.address()}>
-                <MarginTop />
-                <Link to={`/identity/${pair.address()}`}>
-                  <AddressSummary
-                    address={pair.address()}
-                    name={pair.getMeta().name}
-                    orientation='horizontal'
-                    size='small' />
-                </Link>
-              </React.Fragment>
-            );
-          })
-        }
-      </React.Fragment>
-    );
+    return allAccounts && Object.values(allAccounts).map((account: SingleAddress) => {
```

Comment:
This is totally working, but I was thinking about forgetting all the manual subscribes/unsubscribes:
- Jaco uses HOC (which I somehow came to dislike, especially after fether)
- Or I introduced the `<Subscribe />` component in #43, which could be used here:

```javascript
  renderAllAccountsFromKeyring () {
    return (
      <Subscribe>
        {accountObservable.subject.pipe(
          map((allAccounts: SubjectInfo) =>
            Object.values(allAccounts).map((account: SingleAddress) =>
              <React.Fragment key={account.json.address}>
                <MarginTop />
                <Link to={`/identity/${account.json.address}`}>
                  <AddressSummary
                    address={account.json.address}
                    name={account.json.meta.name}
                    orientation='horizontal'
                    size='small'
                  />
                </Link>
              </React.Fragment>
            )
          ))}
      </Subscribe>
    );
  }
```

Reply:
Since you seem to be using `<Subscribe />` everywhere, I would probably lean in that direction.

(hooks & effects are on my TODO list for apps)